spring-content-s3/src/main/java/internal/org/springframework/content/s3/store/DefaultReactiveS3StoreImpl.java

```diff
@@ -0,0 +1,144 @@
+package internal.org.springframework.content.s3.store;
+
+import java.io.File;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.content.commons.mappingcontext.ContentProperty;
+import org.springframework.content.commons.mappingcontext.MappingContext;
+import org.springframework.content.commons.property.PropertyPath;
+import org.springframework.content.commons.repository.ReactiveContentStore;
+import org.springframework.content.commons.repository.StoreAccessException;
+import org.springframework.content.commons.utils.PlacementService;
+import org.springframework.content.s3.S3ObjectId;
+import org.springframework.content.s3.config.MultiTenantS3ClientProvider;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.async.ResponsePublisher;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+@Transactional
+public class DefaultReactiveS3StoreImpl<S, SID extends Serializable>
+		implements ReactiveContentStore<S, SID> {
+
+	private static Log logger = LogFactory.getLog(DefaultReactiveS3StoreImpl.class);
+
+	private ApplicationContext context;
+	private ResourceLoader loader;
+	private PlacementService placementService;
+	private S3Client client;
+	private MultiTenantS3ClientProvider clientProvider;
+
+    private MappingContext mappingContext = new MappingContext(".", ".");
+
+    // reactive
```

Comment:
I think these can be removed. The whole class is reactive

Reply:
The mappingcontext?  Or the comments?